### PR TITLE
Refactor/move args to kwargs

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Move the three common arguments ``dist_sync_on_step``, ``process_group`` and ``dist_sync_fn`` to a ``kwargs`` argument.
The arguments are only meant for real power-users and therefore it does not make sense that they need to be directly available as arguments. This solution is complete backwards compatible and it is therefore more a question about getting more clean documentation as we do not need to repeat theses arguments everywhere.

Currently only updated one metric so we can agree on the refactor before applying it to all other metrics.

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/pages/overview.rst [hunk 1]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
     val = metric.compute() # this value cannot be back-propagated
 
 A functional metric is differentiable if its corresponding modular metric is differentiable.
+
+.. _Metric kwargs:
+
+****************************
+Advance distributed settings
```
Comment (justusschock) [inline docs/source/pages/overview.rst:355]:
```suggestion
Advanced distributed settings
```

File: docs/source/pages/overview.rst [hunk 2]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
...
+****************************
+
+If you are running in a distributed environment, TorchMetrics will automatically take care of the distributed
```
Comment (justusschock) [inline docs/source/pages/overview.rst:358]:
```suggestion
If you are running in a distributed environment, ``TorchMetrics`` will automatically take care of the distributed
```

File: docs/source/pages/overview.rst [hunk 3]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
...
+synchronization for you. However, the following three keyword arguments can be given to any modular metric for
```
Comment (justusschock) [inline docs/source/pages/overview.rst:359]:
```suggestion
synchronization for you. However, the following three keyword arguments can be given to any metric class for
```

I think modular means something different :D

File: docs/source/pages/overview.rst [hunk 4]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
...
+further control over the distributed synchronization:
```
Comment (justusschock) [inline docs/source/pages/overview.rst:360]:
```suggestion
further control over the distributed aggregation:
```

File: docs/source/pages/overview.rst [hunk 5]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
...
+
+- ``dist_sync_on_step``: This argument is ``bool`` that indicates if the metric should syncronize between
+  different devices every time ``forward`` is called. Setting this to ``True`` is in general not recommended
+  as syncronization is an expensive operation to do after each batch.
+
+- ``process_group``: By default we syncronize across the *world* i.e. all proceses being computed on. You
+  can provide an ``torch._C._distributed_c10d.ProcessGroup`` in this argument to specify exactly what
```
Comment (justusschock) [inline docs/source/pages/overview.rst:367]: Is this really the only way to specify this? Also this doesn't do cross-links and if we don't do cross-links we probably don't need the whole path including the private C part of torch, right?

File: torchmetrics/metric.py [hunk 6]
```diff
@@ -70,25 +71,46 @@ class Metric(Module, ABC):
...
+            - dist_sync_fn: function that performs the allgather option on the metric state
     """
 
     __jit_ignored_attributes__ = ["device"]
     __jit_unused_properties__ = ["is_differentiable"]
     is_differentiable: Optional[bool] = None
     higher_is_better: Optional[bool] = None
+    dist_sync_on_step: bool = False
+    process_group: Optional[ProcessGroup] = None
+    dist_sync_fn: Optional[Callable] = None
```
Comment (justusschock) [inline torchmetrics/metric.py:108]: I think we cannot put them here, as these are static. Meaning that This way, I cannot have two instances of the same metric for different process goups or something like that. They need to be moved inside the `__init__`

Comment (SkafteNicki) [reply torchmetrics/metric.py:108]: True. Will change.

File: docs/source/pages/overview.rst (hunk 5 above)
Comment (SkafteNicki) [reply docs/source/pages/overview.rst:367]: I am really not sure, but from the docs it seems so:
https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather

File: docs/source/pages/overview.rst [hunk 7]
```diff
@@ -348,3 +348,24 @@ In practise this means that:
     val = metric.compute() # this value cannot be back-propagated
 
 A functional metric is differentiable if its corresponding modular metric is differentiable.
+
+.. _Metric kwargs:
+
+****************************
+Advanced distributed settings
+****************************
```
Comment (SkafteNicki) [inline docs/source/pages/overview.rst:356]:
```suggestion
*****************************
Advanced distributed settings
*****************************
```

**Review by Borda — COMMENTED** [review]
why do some have `metric_kwargs` and others `metric_kwargs`?

**Comment by SkafteNicki** [comment]
> why do some have `metric_kwargs` and others `metric_kwargs`?

Sorry, my intension was that all should just be `kwargs`. Should be fixed now :]

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Pull request merged**